### PR TITLE
quicker-start with npm install

### PR DESCRIPTION
**Pull request by alexeagle**
We have the prerequisite typings cleanups all merged, and no longer relying on dgeni magic. This PR implements the remaining bits to deliver our typings to users through the `npm install` distribution with no additional commands required.

cc @mprobst @rkirov

### CODE REVIEW

**Comment by alexeagle** [comment]
FYI, reviewed in-person with @mhevery

File: modules/angular2/src/core/linker/view_ref.ts [hunk 1]
```diff
@@ -166,5 +170,10 @@ export class ViewRef_ extends ViewRef {
 export abstract class ProtoViewRef {}
 
 export class ProtoViewRef_ extends ProtoViewRef {
-  constructor(public _protoView: viewModule.AppProtoView) { super(); }
+  /** @internal */
+  public _protoView: viewModule.AppProtoView;
+  constructor(_protoView: viewModule.AppProtoView) {
+    super();
+    this._protoView = _protoView;
+  }
```
Comment (todoubaba) [inline modules/angular2/src/core/linker/view_ref.ts]: Is this pattern still needed? Why not 
```typescript
export class ProtoViewRef {
 /** @internal **/
 _protoView: viewModel.AppProtoView;
 ...
}
```

Comment (alexeagle) [inline modules/angular2/src/core/linker/view_ref.ts]: yes, it is still needed, because we want any instantiation of ProtoViewRef to be a typechecker error

Comment (todoubaba) [inline modules/angular2/src/core/linker/view_ref.ts]: Oh, I see, this pattern relate to injection.

**Pull request closed without merging**